docs/development/design/device-registration.md

```diff
@@ -0,0 +1,47 @@
+# Device registration flow using dynamic client registration in keycloak
+
+> [Issue #1209](https://github.com/nexodus-io/nexodus/issues/1209)
+
+## Summary
+
+* Device registration through access a token
+* Support non-browser authorization flow to allow
+* Client representation for per device in Keycloak
+
+## Proposal
+
+### Problem
+
+Currently, devices are registered on behalf of users using the user and password. If I register 3 devices using the admin user, I will see three sessions against the admin user. It will be challenging to differentiate the device request for authentication and maintain sessions.
+
+### Solution
+
+With the help of the [keycloak dynamic client registration](https://medium.com/keycloak/dynamic-client-registration-in-keycloak-4dd1c5cd5e69) feature, I am proposing to generate an initial access token with a configurable max count and expiration date.
+
+![design-registration flow](./diagrams/design-registration-d1.png)
+
+#### Steps to register device
+
+1. Register a initial access token with a count number and expiration date.
+    1. command: `nexd --username --password generate-intial-token --count 10 (clients) --exp date`
```

Comment:
We use `nexctl` (not `nexd`)  to operate against the control plane.   So I think a command like this would make more sense:
`nexctl registration-token create --count 10 (clients) --exp date`